documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,322 @@
++-----------------------------+----------------------------------------------------------+
+| ``std::uint8_t RadixBits``  | Number of bits to sort per a radix sort algorithm pass.  |
+|                             | Only ``8`` is currently supported.                       |
++-----------------------------+----------------------------------------------------------+
+
+
+Parameters
+----------
+
++------------------------------------------------------+------------------------------------------------------------------+
+| Name                                                 | Description                                                      |
++======================================================+==================================================================+
+|  ``q``                                               | SYCL queue to submit the kernel template to.                     |
++------------------------------------------------------+------------------------------------------------------------------+
+|                                                      | The sequences(s) of elements to apply the algorithm to.          |
+|  - ``rng`` (1)                                       | They can be provided as:                                         |
+|  - ``first``, ``last`` (2)                           |                                                                  |
+|  - ``keys_rng``, ``vals_rng`` (3)                    | - ``sycl::buffer`` (1,3),                                        |
+|  - ``keys_first``, ``keys_last``, ``vals_first`` (4) | - ``oneapi::dpl::experimental::ranges::views::all`` (1,3)        |
+|                                                      | - ``oneapi::dpl::experimental::ranges::views::subrange`` (1,3)   |
+|                                                      | - USM pointer (2,4)                                              |
+|                                                      | - ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` (2,4)          |
++------------------------------------------------------+------------------------------------------------------------------+
+|  ``param``                                           | Kernel configuration structure. ``data_per_workitem``,           |
+|                                                      | can be any value among ``32``, ``64``, ``96``,..., ``k * 32``;   |
+|                                                      | ``workgroup_size`` can be only ``64``.                           |
++------------------------------------------------------+------------------------------------------------------------------+
+
+
+Return Value
+------------
+
+``sycl::event`` object representing a status of the algorithm execution.
+
+
+Memory Usage
+------------
+
+.. _local-memory:
+
+Local Memory
+~~~~~~~~~~~~
+
+The local memory is allocated as shown in the pseudo-code blocks below:
```

Comment:
For clarity I suggest we be more explicit here, and add some verbosity:

"The algorithms require local (SLM) memory, and will allocate this memory according to the following formulas, where `workgroup_size`, and `data_per_workitem` are part of the `kernel_param` structure (link), and `RadixBits`, is a template parameter to the API, and `key_type`, `val_type` are the `value_type` of the input keys, values respectively."

`radix_bits` would need to be `RadixBits` to match the API, and `value_type` would change to `val_type` to remove ambiguity as we describe the `value_type` of certain ranges / iterators.

Reply:
Thanks, I've applied the suggestion